presto-hive-common/src/main/java/com/facebook/presto/hive/zorder/ZOrder.java

```diff
@@ -191,6 +196,23 @@ public long zOrderByteAddressToLong(byte[] byteAddress)
         return byteAddress;
     }
 
+    /**
+     * Searches for and outputs ranges of long addresses within certain ranges in each dimension.
+     */
+    public List<ZAddressRange<Long>> zOrderSearchCurveLongs(List<ZValueRange> ranges)
+    {
+        return curveSearcher.zOrderCondensedSearchCurve(ranges, RangeType.LONG);
+    }
+
+    /**
+     * Searches for and outputs ranges of integer addresses within certain ranges in each dimension.
+     */
+    public List<ZAddressRange<Integer>> zOrderSearchCurveIntegers(List<ZValueRange> ranges)
+    {
+        return curveSearcher.zOrderCondensedSearchCurve(ranges, RangeType.INTEGER).stream()
+                .map(addressRange -> new ZAddressRange<>(addressRange.getMinimumAddress().intValue(), addressRange.getMaximumAddress().intValue())).collect(Collectors.toList());
```

Comment:
I'm wondering if the following refactor makes sense: 

We introduced two methods: `zOrderCondensedSearchCurveForLong` and `zOrderCondensedSearchCurveForInteger`. Internally, they both call `zOrderCondensedSearchCurve` (this becomes private), and `ForInteger` would need to do some post processing to match the return type.

This way, we could hide `RangeType`, so it becomes an internal object for searcher. Also, the caller does not need to handle the post processing.

Reply:
Yes, that was the plan :)
The main goal is to provide public methods that can return either Long or Integer address types without needing the caller to do extra work, while keeping all the internal logic private.